False `import-private-name` positive for relative imports

### Bug description

# pylint: disable=disallowed-name,relative-beyond-top-level,unused-import,missing-module-docstring
from .models import _Private
from ..models import _OtherPrivate


### Configuration

_No response_

### Command used

```shell
pylint --enable-all-extensions foo.py
```


### Pylint output

```shell
************* Module foo
foo.py:2:0: C2701: Imported private object (_Private) (import-private-name)
foo.py:3:0: C2701: Imported private object (_OtherPrivate) (import-private-name)
```


### Expected behavior

No `import-private-name` messages.

I think relative imports should be exempt from this check because they do not tend to refer to external things which are documented and intended to not get this report.

### Pylint version

```shell
pylint 2.14.3
astroid 2.11.6
Python 3.8.12 (default, Sep  4 2021, 14:14:03) 
[GCC 9.3.0]
```


### OS / Environment

_No response_

### Additional dependencies

_No response_

--- a/doc/whatsnew/2/2.14/full.rst
+++ b/doc/whatsnew/2/2.14/full.rst
@@ -18,6 +18,10 @@ Release date: TBA
 
   Closes #6965
 
+* Don't report ``import-private-name`` for relative imports.
+
+  Closes #7078
+
 
 What's New in Pylint 2.14.3?
 ----------------------------

--- a/pylint/extensions/private_import.py
+++ b/pylint/extensions/private_import.py
@@ -250,6 +250,8 @@ class PrivateImportChecker(BaseChecker):
         """Does the node's file's path contain the base name of `import_mod_name`?"""
         if not import_mod_name:  # from . import ...
             return True
+        if node.level:  # from .foo import ..., from ..bar import ...
+            return True
 
         base_import_package = import_mod_name.split(".")[0]
 

--- a/tests/functional/ext/private_import/private_import.py
+++ b/tests/functional/ext/private_import/private_import.py
@@ -1,7 +1,7 @@
 """Tests for import-private-name."""
 # pylint: disable=unused-import, missing-docstring, reimported, import-error, wrong-import-order
 # pylint: disable=no-name-in-module, multiple-imports, ungrouped-imports, misplaced-future
-# pylint: disable=wrong-import-position
+# pylint: disable=wrong-import-position, relative-beyond-top-level
 
 # Basic cases
 from _world import hello  # [import-private-name]
@@ -132,3 +132,8 @@ class Regression6624:
 class Example:
     def save(self):
         return self
+
+
+# Treat relative imports as internal
+from .other_file import _private
+from ..parent import _private

